From: Carsten Schoenert <c.schoenert@t-online.de>
Date: Sat, 30 Jun 2018 20:51:11 +0200
Subject: Don't build testing suites and stuff part 2

Some more testsuites which are not buildable due the cleaning out of
some DFSG non compatible files, mostly HTML (with JS) or JS files.
---
 devtools/client/debugger/new/moz.build             | 18 ++++++------
 devtools/client/shared/moz.build                   |  2 +-
 devtools/client/shared/source-map/moz.build        | 10 ++++---
 devtools/client/shared/vendor/moz.build            | 16 +++++------
 devtools/client/sourceeditor/tern/moz.build        |  2 +-
 .../test/fixtures/stubs/moz.build                  |  6 ++--
 devtools/server/actors/utils/moz.build             |  2 +-
 devtools/shared/acorn/moz.build                    | 12 ++++----
 devtools/shared/sourcemap/moz.build                |  4 ++-
 dom/canvas/moz.build                               | 16 ++++++-----
 dom/html/moz.build                                 |  8 +++---
 dom/indexedDB/moz.build                            | 20 ++++++++------
 dom/media/moz.build                                | 10 ++++---
 dom/push/moz.build                                 | 14 ++++++----
 editor/libeditor/moz.build                         | 10 ++++---
 layout/base/moz.build                              |  2 +-
 services/common/moz.build                          |  6 ++--
 toolkit/components/extensions/moz.build            | 32 ++++++++++++----------
 toolkit/components/passwordmgr/moz.build           | 18 ++++++------
 toolkit/components/prompts/moz.build               |  4 ++-
 toolkit/components/search/moz.build                |  4 ++-
 21 files changed, 122 insertions(+), 94 deletions(-)

diff --git a/devtools/client/debugger/new/moz.build b/devtools/client/debugger/new/moz.build
index ecf13ff..00c97fd 100644
--- a/devtools/client/debugger/new/moz.build
+++ b/devtools/client/debugger/new/moz.build
@@ -3,11 +3,13 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-DevToolsModules(
-    'debugger.css',
-    'debugger.js',
-    'panel.js',
-    'parser-worker.js',
-    'pretty-print-worker.js',
-    'search-worker.js',
-)
+# Don't build any testsuites. We have filteres out some needed content for
+# this as some files are not DFSG clean.
+#DevToolsModules(
+#    'debugger.css',
+#    'debugger.js',
+#    'panel.js',
+#    'parser-worker.js',
+#    'pretty-print-worker.js',
+#    'search-worker.js',
+#)
diff --git a/devtools/client/shared/moz.build b/devtools/client/shared/moz.build
index 5a08e1d..61d3f4e 100644
--- a/devtools/client/shared/moz.build
+++ b/devtools/client/shared/moz.build
@@ -27,7 +27,7 @@ DevToolsModules(
     'browser-loader.js',
     'css-angle.js',
     'curl.js',
-    'demangle.js',
+#    'demangle.js',
     'developer-toolbar.js',
     'devices.js',
     'DOMHelpers.jsm',
diff --git a/devtools/client/shared/source-map/moz.build b/devtools/client/shared/source-map/moz.build
index 21f634c..e5fd35f 100644
--- a/devtools/client/shared/source-map/moz.build
+++ b/devtools/client/shared/source-map/moz.build
@@ -4,7 +4,9 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-DevToolsModules(
-    'index.js',
-    'worker.js',
-)
+# Don't build any testsuites. We have filteres out some needed content for
+# this as some files are not DFSG clean.
+#DevToolsModules(
+#    'index.js',
+#    'worker.js',
+#)
diff --git a/devtools/client/shared/vendor/moz.build b/devtools/client/shared/vendor/moz.build
index 0775cfa..ad82962 100644
--- a/devtools/client/shared/vendor/moz.build
+++ b/devtools/client/shared/vendor/moz.build
@@ -11,21 +11,21 @@ DIRS += [
 DevToolsModules(
     'immutable.js',
     'jsol.js',
-    'jszip.js',
+#    'jszip.js',
     'lodash.js',
     'react-dom-factories.js',
-    'react-dom-server.js',
+#    'react-dom-server.js',
     'react-dom-test-utils.js',
-    'react-dom.js',
-    'react-prop-types.js',
-    'react-redux.js',
-    'react-test-renderer-shallow.js',
+#    'react-dom.js',
+#    'react-prop-types.js',
+#    'react-redux.js',
+#    'react-test-renderer-shallow.js',
     'react.js',
-    'redux.js',
+#    'redux.js',
     'reselect.js',
     'seamless-immutable.js',
     'WasmDis.js',
-    'WasmParser.js',
+#    'WasmParser.js',
 )
 
 # react dev versions are used if either debug mode is enabled,
diff --git a/devtools/client/sourceeditor/tern/moz.build b/devtools/client/sourceeditor/tern/moz.build
index cba3f5a..4553da7 100644
--- a/devtools/client/sourceeditor/tern/moz.build
+++ b/devtools/client/sourceeditor/tern/moz.build
@@ -11,7 +11,7 @@ DevToolsModules(
     'comment.js',
     'condense.js',
     'def.js',
-    'ecma5.js',
+#    'ecma5.js',
     'infer.js',
     'signal.js',
     'tern.js',
diff --git a/devtools/client/webconsole/new-console-output/test/fixtures/stubs/moz.build b/devtools/client/webconsole/new-console-output/test/fixtures/stubs/moz.build
index ae0f9dc..d053f66 100644
--- a/devtools/client/webconsole/new-console-output/test/fixtures/stubs/moz.build
+++ b/devtools/client/webconsole/new-console-output/test/fixtures/stubs/moz.build
@@ -3,10 +3,10 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 DevToolsModules(
-  'consoleApi.js',
+#  'consoleApi.js',
   'cssMessage.js',
-  'evaluationResult.js',
+#  'evaluationResult.js',
   'index.js',
   'networkEvent.js',
-  'pageError.js',
+#  'pageError.js',
 )
diff --git a/devtools/server/actors/utils/moz.build b/devtools/server/actors/utils/moz.build
index 45f4e57..26ba1ae 100644
--- a/devtools/server/actors/utils/moz.build
+++ b/devtools/server/actors/utils/moz.build
@@ -7,7 +7,7 @@
 DevToolsModules(
     'actor-registry-utils.js',
     'audionodes.json',
-    'automation-timeline.js',
+#    'automation-timeline.js',
     'breakpoint-actor-map.js',
     'css-grid-utils.js',
     'event-loop.js',
diff --git a/devtools/shared/acorn/moz.build b/devtools/shared/acorn/moz.build
index 8fd5375..78c5701 100644
--- a/devtools/shared/acorn/moz.build
+++ b/devtools/shared/acorn/moz.build
@@ -6,8 +6,10 @@
 
 XPCSHELL_TESTS_MANIFESTS += ['tests/unit/xpcshell.ini']
 
-DevToolsModules(
-    'acorn.js',
-    'acorn_loose.js',
-    'walk.js',
-)
+# Don't build any testsuites. We have filteres out some needed content for
+# this as some files are not DFSG clean.
+#DevToolsModules(
+#    'acorn.js',
+#    'acorn_loose.js',
+#    'walk.js',
+#)
diff --git a/devtools/shared/sourcemap/moz.build b/devtools/shared/sourcemap/moz.build
index 4597496..0ee9931 100644
--- a/devtools/shared/sourcemap/moz.build
+++ b/devtools/shared/sourcemap/moz.build
@@ -4,7 +4,9 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-XPCSHELL_TESTS_MANIFESTS += ['tests/unit/xpcshell.ini']
+# Don't build any testsuites. We have filteres out some needed content for
+# this as some files are not DFSG clean.
+#XPCSHELL_TESTS_MANIFESTS += ['tests/unit/xpcshell.ini']
 
 DevToolsModules(
     'source-map.js'
diff --git a/dom/canvas/moz.build b/dom/canvas/moz.build
index e1bd521..1e4b81d 100644
--- a/dom/canvas/moz.build
+++ b/dom/canvas/moz.build
@@ -35,13 +35,15 @@ TEST_DIRS += [
 # Change the following line(s) to avoid bug 1081323 (clobber after changing a manifest):
 # * Adjust failure errata for webgl-conf.
 
-MOCHITEST_MANIFESTS += [
-    'test/crash/mochitest.ini',
-    'test/crossorigin/mochitest.ini',
-    'test/mochitest.ini',
-    'test/webgl-conf/generated-mochitest.ini',
-    'test/webgl-mochitest/mochitest.ini',
-]
+# Don't build any testsuites. We have filteres out some needed content for
+# this as some files are not DFSG clean.
+#MOCHITEST_MANIFESTS += [
+#    'test/crash/mochitest.ini',
+#    'test/crossorigin/mochitest.ini',
+#    'test/mochitest.ini',
+#    'test/webgl-conf/generated-mochitest.ini',
+#    'test/webgl-mochitest/mochitest.ini',
+#]
 
 MOCHITEST_CHROME_MANIFESTS += ['test/chrome/chrome.ini']
 
diff --git a/dom/html/moz.build b/dom/html/moz.build
index 93551d6..ca70c3e 100644
--- a/dom/html/moz.build
+++ b/dom/html/moz.build
@@ -9,10 +9,10 @@ with Files("**"):
 
 DIRS += ['input']
 
-MOCHITEST_MANIFESTS += [
-    'test/forms/mochitest.ini',
-    'test/mochitest.ini',
-]
+#MOCHITEST_MANIFESTS += [
+#    'test/forms/mochitest.ini',
+#    'test/mochitest.ini',
+#]
 
 MOCHITEST_CHROME_MANIFESTS += [
     'test/chrome.ini',
diff --git a/dom/indexedDB/moz.build b/dom/indexedDB/moz.build
index 3f9f46d..f6b0e39 100644
--- a/dom/indexedDB/moz.build
+++ b/dom/indexedDB/moz.build
@@ -7,19 +7,23 @@
 with Files("**"):
     BUG_COMPONENT = ("Core", "DOM: IndexedDB")
 
-MOCHITEST_MANIFESTS += [
-    'test/mochitest-intl-api.ini',
-    'test/mochitest.ini',
-]
+# Don't build any testsuites. We have filteres out some needed content for
+# this as some files are not DFSG clean.
+#MOCHITEST_MANIFESTS += [
+#    'test/mochitest-intl-api.ini',
+#    'test/mochitest.ini',
+#]
 
 BROWSER_CHROME_MANIFESTS += ['test/browser.ini']
 
 MOCHITEST_CHROME_MANIFESTS += ['test/chrome.ini']
 
-XPCSHELL_TESTS_MANIFESTS += [
-    'test/unit/xpcshell-child-process.ini',
-    'test/unit/xpcshell-parent-process.ini'
-]
+# Don't build any testsuites. We have filteres out some needed content for
+# this as some files are not DFSG clean.
+#XPCSHELL_TESTS_MANIFESTS += [
+#    'test/unit/xpcshell-child-process.ini',
+#    'test/unit/xpcshell-parent-process.ini'
+#]
 
 EXPORTS.mozilla.dom += [
     'IDBCursor.h',
diff --git a/dom/media/moz.build b/dom/media/moz.build
index 4b1ca53..d60d4ad 100644
--- a/dom/media/moz.build
+++ b/dom/media/moz.build
@@ -68,10 +68,12 @@ TEST_DIRS += [
     'gtest',
 ]
 
-MOCHITEST_MANIFESTS += [
-    'test/mochitest.ini',
-    'tests/mochitest/identity/mochitest.ini',
-]
+# Don't build any testsuites. We have filteres out some needed content for
+# this as some files are not DFSG clean.
+#MOCHITEST_MANIFESTS += [
+#    'test/mochitest.ini',
+#    'tests/mochitest/identity/mochitest.ini',
+#]
 
 if CONFIG['MOZ_WEBRTC']:
     MOCHITEST_MANIFESTS += ['tests/mochitest/mochitest.ini']
diff --git a/dom/push/moz.build b/dom/push/moz.build
index f1aba80..3744330 100644
--- a/dom/push/moz.build
+++ b/dom/push/moz.build
@@ -31,13 +31,15 @@ else:
         'PushServiceAndroidGCM.jsm',
     ]
 
-MOCHITEST_MANIFESTS += [
-    'test/mochitest.ini',
-]
+# Don't build any testsuites. We have filteres out some needed content for
+# this as some files are not DFSG clean.
+#MOCHITEST_MANIFESTS += [
+#    'test/mochitest.ini',
+#]
 
-XPCSHELL_TESTS_MANIFESTS += [
-    'test/xpcshell/xpcshell.ini',
-]
+#XPCSHELL_TESTS_MANIFESTS += [
+#    'test/xpcshell/xpcshell.ini',
+#]
 
 EXPORTS.mozilla.dom += [
     'PushManager.h',
diff --git a/editor/libeditor/moz.build b/editor/libeditor/moz.build
index f7b2520..f35a5e0 100644
--- a/editor/libeditor/moz.build
+++ b/editor/libeditor/moz.build
@@ -4,10 +4,12 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-MOCHITEST_MANIFESTS += [
-    'tests/browserscope/mochitest.ini',
-    'tests/mochitest.ini',
-]
+# Don't build any testsuites. We have filteres out some needed content for
+# this as some files are not DFSG clean.
+#MOCHITEST_MANIFESTS += [
+#    'tests/browserscope/mochitest.ini',
+#    'tests/mochitest.ini',
+#]
 
 MOCHITEST_CHROME_MANIFESTS += ['tests/chrome.ini']
 
diff --git a/layout/base/moz.build b/layout/base/moz.build
index 35fbcf2..82628c7 100644
--- a/layout/base/moz.build
+++ b/layout/base/moz.build
@@ -175,7 +175,7 @@ FINAL_LIBRARY = 'xul'
 
 BROWSER_CHROME_MANIFESTS += ['tests/browser.ini']
 MARIONETTE_LAYOUT_MANIFESTS += ['tests/marionette/manifest.ini']
-MOCHITEST_MANIFESTS += ['tests/mochitest.ini']
+#MOCHITEST_MANIFESTS += ['tests/mochitest.ini']
 MOCHITEST_CHROME_MANIFESTS += ['tests/chrome/chrome.ini']
 
 if CONFIG['CC_TYPE'] in ('msvc', 'clang-cl'):
diff --git a/services/common/moz.build b/services/common/moz.build
index c5a8cbd..1937df7 100644
--- a/services/common/moz.build
+++ b/services/common/moz.build
@@ -17,9 +17,9 @@ EXTRA_JS_MODULES['services-common'] += [
     'async.js',
     'blocklist-clients.js',
     'blocklist-updater.js',
-    'kinto-http-client.js',
-    'kinto-offline-client.js',
-    'kinto-storage-adapter.js',
+#    'kinto-http-client.js',
+#    'kinto-offline-client.js',
+#    'kinto-storage-adapter.js',
     'logmanager.js',
     'observers.js',
     'rest.js',
diff --git a/toolkit/components/extensions/moz.build b/toolkit/components/extensions/moz.build
index 2b0c1a2..22f49b9 100644
--- a/toolkit/components/extensions/moz.build
+++ b/toolkit/components/extensions/moz.build
@@ -74,21 +74,23 @@ FINAL_LIBRARY = 'xul'
 
 JAR_MANIFESTS += ['jar.mn']
 
-BROWSER_CHROME_MANIFESTS += [
-    'test/browser/browser.ini',
-]
-
-MOCHITEST_MANIFESTS += [
-    'test/mochitest/mochitest-remote.ini',
-    'test/mochitest/mochitest.ini'
-]
-MOCHITEST_CHROME_MANIFESTS += ['test/mochitest/chrome.ini']
-XPCSHELL_TESTS_MANIFESTS += [
-    'test/xpcshell/native_messaging.ini',
-    'test/xpcshell/xpcshell-e10s.ini',
-    'test/xpcshell/xpcshell-remote.ini',
-    'test/xpcshell/xpcshell.ini',
-]
+# Don't build any testsuites. We have filteres out some needed content for
+# this as some files are not DFSG clean.
+#BROWSER_CHROME_MANIFESTS += [
+#    'test/browser/browser.ini',
+#]
+
+#MOCHITEST_MANIFESTS += [
+#    'test/mochitest/mochitest-remote.ini',
+#    'test/mochitest/mochitest.ini'
+#]
+#MOCHITEST_CHROME_MANIFESTS += ['test/mochitest/chrome.ini']
+#XPCSHELL_TESTS_MANIFESTS += [
+#    'test/xpcshell/native_messaging.ini',
+#    'test/xpcshell/xpcshell-e10s.ini',
+#    'test/xpcshell/xpcshell-remote.ini',
+#    'test/xpcshell/xpcshell.ini',
+#]
 
 SPHINX_TREES['webextensions'] = 'docs'
 
diff --git a/toolkit/components/passwordmgr/moz.build b/toolkit/components/passwordmgr/moz.build
index 72c8c70..bb9de59 100644
--- a/toolkit/components/passwordmgr/moz.build
+++ b/toolkit/components/passwordmgr/moz.build
@@ -7,16 +7,18 @@
 if CONFIG['MOZ_BUILD_APP'] == 'browser':
     DEFINES['MOZ_BUILD_APP_IS_BROWSER'] = True
 
-MOCHITEST_MANIFESTS += ['test/mochitest.ini', 'test/mochitest/mochitest.ini']
-MOCHITEST_CHROME_MANIFESTS += ['test/chrome/chrome.ini']
-BROWSER_CHROME_MANIFESTS += ['test/browser/browser.ini']
-XPCSHELL_TESTS_MANIFESTS += ['test/unit/xpcshell.ini']
+# Don't build any testsuites. We have filteres out some needed content for
+# this as some files are not DFSG clean.
+#MOCHITEST_MANIFESTS += ['test/mochitest.ini', 'test/mochitest/mochitest.ini']
+#MOCHITEST_CHROME_MANIFESTS += ['test/chrome/chrome.ini']
+#BROWSER_CHROME_MANIFESTS += ['test/browser/browser.ini']
+#XPCSHELL_TESTS_MANIFESTS += ['test/unit/xpcshell.ini']
 
-TESTING_JS_MODULES += [
+#TESTING_JS_MODULES += [
     # Make this file available from the "resource:" URI of the test environment.
-    'test/browser/form_basic.html',
-    'test/LoginTestUtils.jsm',
-]
+#    'test/browser/form_basic.html',
+#    'test/LoginTestUtils.jsm',
+#]
 
 XPIDL_SOURCES += [
     'nsILoginInfo.idl',
diff --git a/toolkit/components/prompts/moz.build b/toolkit/components/prompts/moz.build
index 63b8ba3..7bfd6c3 100644
--- a/toolkit/components/prompts/moz.build
+++ b/toolkit/components/prompts/moz.build
@@ -10,6 +10,8 @@ with Files('**'):
 
 DIRS += ['src']
 
-MOCHITEST_MANIFESTS += ['test/mochitest.ini']
+# Don't build any testsuites. We have filteres out some needed content for
+# this as some files are not DFSG clean.
+#MOCHITEST_MANIFESTS += ['test/mochitest.ini']
 
 JAR_MANIFESTS += ['jar.mn']
diff --git a/toolkit/components/search/moz.build b/toolkit/components/search/moz.build
index 98ccf2b..31c8a72 100644
--- a/toolkit/components/search/moz.build
+++ b/toolkit/components/search/moz.build
@@ -4,7 +4,9 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-XPCSHELL_TESTS_MANIFESTS += ['tests/xpcshell/xpcshell.ini']
+# Don't build any testsuites. We have filteres out some needed content for
+# this as some files are not DFSG clean.
+#XPCSHELL_TESTS_MANIFESTS += ['tests/xpcshell/xpcshell.ini']
 
 EXTRA_COMPONENTS += [
     'nsSearchService.js',
